model_interfaces/charlie_interface.py

```diff
@@ -1,143 +1,149 @@
-        browsers = [
-            browser_cookie3.chrome,
-            browser_cookie3.firefox,
-            browser_cookie3.chromium,
-            browser_cookie3.edge,
-            browser_cookie3.safari,
-        ]
-
-        # Extract session token from cookie
-        # We don't know which browser the user is using, so search for the most obvious ones:
-        for b in browsers:
-            try:
-                cj = list(b(domain_name="clang.goodai.com"))
-                self.token, self.user_name = try_extract_session_cookie(cj)
-                if self.token != "" and self.user_name != "":
-                    break
-            except:
-                continue
-
-        if self.token == "":
-            raise ValueError("No valid clang login found! Please login via browser.")
-
-        body = {"username": self.user_name, "session_token": self.token}
-        valid = requests.post(self.endpoint + "/check_token/", json=body)
-
-        if valid.status_code != 200:
-            raise ValueError(
-                "Username/session token combination found by invalid! Please make sure your cookies are up to date."
-            )
-
-        # Get display name and current costs of user
-        settings_dict = self.get_settings()
-        self.display_name = settings_dict["display_name"][0]
-        self.initial_costs_usd = settings_dict["usage"]["total_cost"]
-
-        # Update max_tokens
-        headers = {
-            "Content-Type": "application/json",
-            "Cookie": f"session_token={self.token}",
+        self.session = requests.Session()
+        self.login()
+        self.create_chat_tab()
+        self.update_memory_settings()
+        self.update_max_tokens()
+        self.load_initial_settings()
+        self.update_system_prompt(self.system_prompt)
+
+    def login(self):
+        login_url = f"{self.endpoint}/login/"
+        login_data = {"username": self.username, "password": self.password}
+        response = self.session.post(login_url, json=login_data)
+        if response.status_code != 200:
+            raise ValueError("Login failed. Please check your credentials.")
+
+    def update_system_prompt(self, new_system_prompt: str):
+        update_settings_url = f"{self.endpoint}/update_settings/"
+        update_data = {
+            "username": self.username,
+            "category": "system_prompt",
+            "setting": {"system_prompt": new_system_prompt},
         }
-        body = {
-            "username": self.user_name,
+        response = self.session.post(update_settings_url, json=update_data)
+        if response.status_code != 200:
+            raise ValueError("Failed to update system prompt setting.")
+        self.system_prompt = new_system_prompt
+
+    def create_chat_tab(self):
+        create_chat_tab_url = f"{self.endpoint}/create_chat_tab/"
+        chat_tab_data = {
+            "username": self.username,
+            "chat_id": self.chat_id,
+            "chat_name": f"{self.chat_id}",
```

Comment:
I wonder if there's any motivation behind the use of the f-string, and what would happen if a chat with the same ID already exists.

Reply:
f-string is not per > I wonder if there's any motivation behind the use of the f-string
F-string is not really needed here, initially I did something in the line of `"chat_name": f"Benchmark {self.chat_id}",`
but then removed the 'Bechmark ' part.

> and what would happen if a chat with the same ID already exists.
Would throw an error, I'll add something to avoid this error.se needed here,